src/v2i-hub/PedestrianPlugin/src/PedestrianPlugin.cpp

```diff
@@ -63,6 +99,57 @@ void PedestrianPlugin::PedestrianRequestHandler(QHttpEngine::Socket *socket)
 	}
 }
 
+int PedestrianPlugin::StartWebSocket()
+{
+	PLOG(logDEBUG) << "In PedestrianPlugin::StartWebSocket " << std::endl;
+	// The io_context is required for all I/O
+    net::io_context ioc;
+
+	flirSession = std::make_shared<FLIRWebSockAsyncClnSession>(ioc);
+
+    // Launch the asynchronous operation
+	flirSession->run(webSocketIP.c_str(), webSocketURLExt.c_str(), cameraRotation);	
+
+	PLOG(logDEBUG) << "Successfully running the I/O service" << std::endl;	
+
+    // Run the I/O service. The call will return when
+    // the socket is closed.
+    ioc.run();
+
+	return EXIT_SUCCESS;
+}
+
+int PedestrianPlugin::checkXML()
+{
+	//first xml will be empty string
+	std::string lastGeneratedXML = "";
+
+	//if a new psm xml has been generated the FLIR web socket, send it to the BroadcastPSM function
+	while (true)
+	{
+		if (flirSession == nullptr)
+		{
+			PLOG(logDEBUG) << "flir session not yet initialized: " << std::endl;
+		}
+		else
+		{
+			std::string currentXML = flirSession->getPSMXML();
+
+			if (currentXML.compare(lastGeneratedXML) != 0)
+			{		
+				PLOG(logINFO) << "Last xml: " << lastGeneratedXML.c_str() << std::endl;
```

Comment:
You may want to consider adjusting the loglevel on some of these statements. **INFO** is usually default so if you are printing a lot of logs, repeatedly this is something to consider.

Reply:
Yes, this was code leftover from my testing. Removed in my latest commit.